docs/proposals/multi-template-partitionting-and-scheduler-extensibility/README.md

```diff
@@ -0,0 +1,603 @@
+
+**I believe this is beneficial for Karmada** as it can enrich the scenarios supported by Karmada and improve its flexibility.
+Currently, if users have their own scheduling strategies, they must modify Karmada's core API, which is a very complex method.
+Implementing through extension points is simpler, representing an Out of Tree approach.
+
+Relevant contents:
+- https://github.com/karmada-io/karmada/issues/3318
+- https://github.com/karmada-io/karmada/issues/3318#issuecomment-1682519228
+- https://docs.google.com/document/d/1l6zO4xf879KdW_WPS7aMED0SUmnk487_XDsC12TtuTQ/edit?disco=AAAAv4FJT8E
+
+## Summary
+
+<!--
+当前的 Karmada Scheduler 支持在 FilterCluster 和 ScoreCluster 阶段注册插件，但在 SelectCluster 和
+ReplicaScheduling (AssignReplicas) 阶段则不支持。我们希望能在 ReplicaScheduling (AssignReplicas) 阶段支持注册自定义插件，以提升 Karmada 的灵活性。
+
+我们将提供一种方案，使得在 ReplicaScheduling (AssignReplicas) 阶段能够支持注册自定义插件。同时包括新的在PropagationPolicy中选择
+插件/策略的方式，以及插件的管理策略。
+-->
+
+The current Karmada Scheduler supports plugin registration during the **FilterCluster** and **ScoreCluster** stages, but not during the **SelectCluster** and  
+**ReplicaScheduling (AssignReplicas)** stages. We aim to enable custom plugin registration during the **ReplicaScheduling (AssignReplicas)** stage to enhance Karmada's flexibility.
+
+![schedule-process](statics/schedule-process.png)
+
+We will provide a solution that allows for the registration of custom plugins during the **ReplicaScheduling (AssignReplicas)** stage.
+This includes methods for handling multiple results from multiple plugins and strategies for plugin management.
+
+### Goals
+
+- Support custom plugins in the `AssignReplicas` stage.
+- Allow users to add custom strategies and select custom strategies in the PropagationPolicy.
+- Properly manage custom plugins contributed by users to the community.
+- Provide design ideas for multi-template splitting.
+
+## Proposal
+
+### User Story
+
+#### Story 1: Precise Cluster Scheduling for AI Tasks
+
+<!--
+作为一个用户，我希望能够将 Karmada 作为一个 AI 任务的管理平台，它需要将**单个 AI 大数据任务**下发到**某一个 Worker Cluster**，
+同时它应当能够**支持 gang 调度**，避免集群资源不足时仍然下发任务导致资源浪费。
+
+在运行 AI 大数据任务时，由于地域、设备、集群大小等限制，我们常常需要为**每个任务选择合适的 Worker Cluster** 来完成它。
+在多集群上，**Karmada 无疑是最好的选择**，但是目前 Scheduler 拓展性的不足导致我们无法做到 gang 调度来精确的选择作业集群。
+-->
+
+As a user, I want to use Karmada as a management platform for AI tasks,
+where it needs
+to distribute **single AI big data tasks** to **a specific Worker Cluster**
+while supporting **gang scheduling**
+to avoid wasting resources by issuing tasks when the cluster resources are insufficient.
+
+During the execution of AI big data tasks, due to limitations such as region, equipment,
+and cluster size, we often need to select the appropriate Worker Cluster for **each task**.
+Among multiple clusters, **Karmada is undoubtedly the best choice**,
+but currently,
+the lack of scalability in the Scheduler prevents us from achieving gang scheduling to precisely select the job cluster.
```

Comment:
Karmada already supports some kind of `gang scheduling` now, [see the kubeflow.PytorchJob example](#4049).

Reply:
Yes, but this method has limitations and is not elegant enough. We hope to better fulfill similar requirements